### PR TITLE
Improve room switching performance with alias cache

### PR DESCRIPTION
**Pull request by jryans**
This adds very basic cache (literally just a `Map` for now) to store room alias
to room ID mappings. The improves the perceived performance of Riot when
switching rooms via browser navigation (back / forward), as we no longer try to
resolve the room alias every time.

The cache is only in memory, so reloading manually or as part of the clear cache
process will start afresh.

Fixes https://github.com/vector-im/riot-web/issues/10020

### CODE REVIEW

File: src/stores/RoomViewStore.js [hunk 1]
```diff
@@ -176,6 +177,22 @@ class RoomViewStore extends Store {
                 this._joinRoom(payload);
             }
         } else if (payload.room_alias) {
+            // Try the room alias to room ID navigation cache first to avoid
+            // blocking room navigation on the homeserver.
+            const roomId = getCachedRoomIDForAlias(payload.room_alias);
```
Comment (t3chguy) [inline src/stores/RoomViewStore.js]: how about flipping this around to be

```suggestion
            let roomId = getCachedRoomIDForAlias(payload.room_alias);
            if (!roomId) {
                try {
                    const result = await MatrixClientPeg.get().getRoomIdForAlias(payload.room_alias);
                    storeRoomAliasInCache(payload.room_alias, result.room_id);
                    roomId = result.room_id;
                } catch (err) {...}
                
                dis.dispatch({...});
```

to not have two identical dispatches next to each other on diff branches

Comment (jryans) [reply src/stores/RoomViewStore.js]: Okay, seems fine to me. (Thought about it before, but couldn't decide what I preferred...)

Comment (jryans) [reply src/stores/RoomViewStore.js]: Okay, seems fine to me. (Thought about it before, but couldn't decide what I preferred...)

**Comment by ofek** [comment]
Thank you!!!!!

**Pull request merged**